eventsdiscovery/src/main/java/com/schedjoules/client/eventsdiscovery/http/MultiEventsResponseHandler.java

```diff
@@ -69,107 +70,94 @@
             @Override
             public Iterable<Envelope<Event>> items()
             {
-                return new Iterable<Envelope<Event>>()
+                return new Mapped<>(new JsonArrayIterable(jsonArray), new Function<JSONObject, Envelope<Event>>()
```

Comment:
I see the class is getting cleaner at some places but still it's still too big to easily read through. Maybe extracting classes to top level? Maybe starting from the whole `ResultPage`?
Not important, of course.